front/src/modules/timesStops/TimesStopsInput.tsx

```diff
@@ -66,24 +58,68 @@ const TimesStopsinput = ({ allWaypoints, startTime, pathSteps }: TimesStopsInput
     dispatch(updatePathSteps({ pathSteps: updatedPathSteps }));
   };
 
+  const onChange = (
+    rows: TimesStopsInputRow[],
+    previousRows: TimesStopsInputRow[],
+    op: Operation,
+    setRows: React.Dispatch<React.SetStateAction<TimesStopsInputRow[]>>
+  ) => {
+    const row = { ...rows[op.fromRowIndex] };
+    const previousRow = { ...previousRows[op.fromRowIndex] };
+    if (
+      row.departure &&
+      row.arrival &&
+      (row.arrival !== previousRow.arrival || row.departure !== previousRow.departure)
+    ) {
+      row.stopFor = String(time2sec(row.departure) - time2sec(row.arrival));
+    }
+    if (!row.stopFor && op.fromRowIndex !== rows.length - 1) {
+      row.onStopSignal = false;
+    }
+
+    if (row.theoreticalMargin && !marginRegExValidation.test(row.theoreticalMargin!)) {
+      row.isMarginValid = false;
+      setRows(rows);
```

Comment:
if we can a way to keep the setRows inside `TimesStops` that would work. but that’s not good design to allow users to know implementation details of an api.
we could call setRows inside `TimesStops` no matter what, and update the store on conditions.

I made a lot of changes here, in #8334 I’d prefer to do this refacto after it’s done